wolfcrypt/src/ed25519.c

```diff
@@ -1028,25 +1051,39 @@ int wc_ed25519_import_private_only(const byte* priv, word32 privSz,
     if (privSz < ED25519_KEY_SIZE)
         return BAD_FUNC_ARG;
 
+    if (key->pubKeySet) {
+        /* public key should not be set when importing a private only.
+         * The public key is used with private key during a signing operation.
+         * For the public to be trusted it either needs to be computed
+         * automatically during a sign or imported after private. If importing
+         * both public and private use wc_ed25519_import_private_key_ex */
+        return BAD_FUNC_ARG;
+    }
```

Comment:
@dgarske how much effort do you think it would be to generate the public key and compare it to the loaded key instead of just outright not allowing it?

Reply:
Very easy. I'll post the change. I could also add a strict build option to error on this behavior.

Reply:
Fixed

Reply:
Can confirm fixed.